src/NServiceBus.Core/Persistence/PersistenceStartup.cs

```diff
@@ -17,12 +18,13 @@ public void Run(SettingsHolder settings)
             }
 
             var enabledPersistences = PersistenceStorageMerger.Merge(definitions, settings);
+            var bothFeaturesEnabled = settings.IsFeatureEnabled(typeof(Features.Sagas)) && settings.IsFeatureEnabled(typeof(Features.Outbox));
+
+            ValidateSagaAndOutboxUseSamePersistence(enabledPersistences, bothFeaturesEnabled);
```

Comment:
I would move the code from line 21 to the `ValidateSagaAndOutboxUseSamePersistence` method. What do you think @timbussmann ?

Reply:
agreed and fixed